### PR TITLE
Limit bugfix

### PR DESCRIPTION
**Pull request by ghost**

### CODE REVIEW

File: src/RegisterMapping.cpp [hunk 1]
```diff
@@ -204,17 +204,17 @@ const std::vector<int16_t>& RegisterMapping::getAddresses() const
...
 }
 
-void RegisterMapping::setSlaveAddress(int8_t slaveAddress)
+void RegisterMapping::setSlaveAddress(int16_t slaveAddress)
 {
     m_slaveAddress = slaveAddress;
 }
 
-int16_t RegisterMapping::getStartingAddress() const
+int32_t RegisterMapping::getStartingAddress() const
```
Comment (srdjastankovic) [inline src/RegisterMapping.cpp:217]: Why this is double in size?

**Comment by srdjastankovic** [comment]
Why  name space RegisterMapping::RegisterMapping(const std::string& reference, RegisterMapping::RegisterType registerType, int16_t address, bool readRestricted, int8_t slaveAddress)
 has parameter type int8_t slaveAddress ?

this file: 
https://github.com/Wolkabout/MoreModbus/blob/31f57b81aeedc633dfe06739f09e574486319935/src/RegisterMapping.cpp

**Comment by srdjastankovic** [comment]
and this:

BoolMapping::BoolMapping(const std::string& reference, RegisterMapping::RegisterType registerType, int16_t address, bool readRestricted, int8_t slaveAddress)

**Comment by srdjastankovic** [comment]
ModbusDevice::ModbusDevice(const std::string& name, int8_t slaveAddress)
: m_name(name), m_status(false), m_slaveAddress(slaveAddress), m_groups()
{
}

**Comment by srdjastankovic** [comment]
UInt16Mapping::UInt16Mapping(const std::string& reference, RegisterMapping::RegisterType registerType, int16_t address,  bool readRestricted, int8_t slaveAddress)

**Comment by srdjastankovic** [comment]
Int16Mapping::Int16Mapping(const std::string& reference, RegisterMapping::RegisterType registerType, int16_t address, bool readRestricted, int8_t slaveAddress)

**Comment by srdjastankovic** [comment]
MOCK_METHOD0(getSlaveAddress, int8_t());
    MOCK_METHOD1(setSlaveAddress, void(int8_t));

**Comment by srdjastankovic** [comment]
bool readRestricted = false, int8_t slaveAddress = -1);

**Comment by srdjastankovic** [comment]
OperationType operation, bool readRestricted = false, int8_t slaveAddress = -1);

**Comment by srdjastankovic** [comment]
StringMapping(const std::string& reference, RegisterType registerType, const std::vector<int16_t>& addresses,
                  OperationType operation, bool readRestricted = false, int8_t slaveAddress = -1);

**Comment by srdjastankovic** [comment]
OperationType operation, bool readRestricted = false, int8_t slaveAddress = -1);

File: src/RegisterMapping.cpp (hunk 1 above)
Comment (ghost) [reply src/RegisterMapping.cpp:217]: Because, if you enter your address 35003 for example, it will become -30533.

Comment (srdjastankovic) [reply src/RegisterMapping.cpp:217]: ok this is reg add, not slave add

**Review by milanzlicic — APPROVED** [review]

**Pull request merged**